### ISSUE TITLE
feat(optimizers): Add Adadelta — adaptive LR without global learning rate

### ISSUE THREAD

**Issue by sanskarpan** [sanskarpan/Optimization#28]
## Description
Implement **Adadelta** (Zeiler, 2012) which eliminates the need for a global learning rate.

## Algorithm
```
E[g²] = ρ·E[g²] + (1-ρ)·g²
Δx = -(RMS[Δx] / RMS[g]) · g
E[Δx²] = ρ·E[Δx²] + (1-ρ)·Δx²
θ += Δx
```

## Acceptance Criteria
- [ ] No explicit `learning_rate` parameter (rho and eps only)
- [ ] RMS ratio drives step size
- [ ] `get_state()`, `load_state()`, `__repr__()`
- [ ] Tests: no-LR usage, RMS ratio, convergence

**Comment by sanskarpan** [comment]
Resolved in PR #69 (merged). Adadelta implemented in `optimizers.py` with RMS ratio of delta/gradient, no global learning rate.
